gateway/src/resty/oidc/discovery.lua

```diff
@@ -40,10 +46,41 @@ function _M.new(http_backend)
             ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') }
         }
     }
+    return _M.new_with_http_client(http_client)
+end
+
+function _M.new_with_http_client(http_client)
+    local self = setmetatable({ http_client = http_client }, mt)
+    self:init_cache()
+    return self
+end
 
-    local self = { http_client = http_client }
 
-    return setmetatable(self, mt)
+function _M:init_cache()
+  self.cache = resty_lrucache.new(cache_max_size)
+  return self
+end
+
+
+--- Fetch and return OIDC configuration from cache if it exists.
+function _M:issuer_in_cache(issuer)
+  if not self.cache then
+    return nil
+  end
+
+  return self.cache:get(issuer)
+end
+
+--- Insert the OIDC configuration for the issuer in the local cache with a
+--limit. If no ttl set will be 0
+function _M:save_issuer_in_cache(issuer, config, ttl)
+  local expires = tonumber(ttl) or 0
+
+  if not self.cache then
```

Comment:
Nitpick : I think it's better to move this `if` to be the first line of the function.
I think it clearly says that if there's no cache there's nothing else to do.